src/Lecture2.hs

```diff
@@ -48,7 +50,10 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct list = case list of
+    [] -> 1
+    (0:_) -> 0
+    (x:xs) -> x * lazyProduct(xs)
```

Comment:
That's a great implementation! 🔝 
Only one minor suggestion that `()` are redundant here
```suggestion
    0 : _ -> 0
    x : xs -> x * lazyProduct xs
```